packages/components/src/components/static-url/static-url.jsx

```diff
@@ -1,8 +1,10 @@
 import React from 'react';
-import { getStaticUrl, PlatformContext } from '@deriv/shared';
+import { getStaticUrl, PlatformContext, setUrlLanguage } from '@deriv/shared';
+import { getLanguage } from '@deriv/translations';
 
 const StaticUrl = ({ href, is_document, children, ...props }) => {
     const { is_appstore } = React.useContext(PlatformContext);
+    setUrlLanguage(getLanguage());
```

Comment:
we need to call this func synchronously, as getStaticUrl func uses language variable in the same scope

Reply:
```suggestion
    const getHref = () => {
        setUrlLanguage(getLanguage());
        return getStaticUrl(href, { is_appstore }, is_document);
    }
```
this way we'll set a correct language right at the moment when we click this link 😊 please check if it works)

Reply:
Actually, this is one of my variants for implementation :) . But on the whole it is the same synchronous code, which is executed with rendering.
So there are no differences :)
But I like this as it looks like more clear.  will implement :+1: